src/server/pkg/collection/transaction.go

```diff
@@ -206,9 +222,28 @@ func (s *stm) Put(key, val string, ttl int64, ptr uintptr) error {
 	return nil
 }
 
-func (s *stm) Del(key string) { s.wset[key] = stmPut{"", 0, v3.OpDelete(key), 0} }
+func (s *stm) Del(key string) {
+	s.wset[key] = stmPut{"", 0, v3.OpDelete(key), 0}
+}
 
-func (s *stm) DelAll(key string) { s.wset[key] = stmPut{"", 0, v3.OpDelete(key, v3.WithPrefix()), 0} }
+func (s *stm) DelAll(prefix string) {
+	// Remove any eclipsed deletes then add the new delete
```

Comment:
What happens if you have pass a prefix that is eclipsed by a pre-existing deleted prefix? Is that not an issue?

Reply:
That would result in a `duplicate key given in txn request` - so we should probably just filter those out, good call.  I'll add a test.